packages/api-queries/src/site.ts

```diff
@@ -15,8 +15,15 @@ import type { Query } from '@tanstack/react-query';
 
 const KNOWN_ERRORS = [ 'unknown_blog', 'unauthorized' ];
 
-export const siteBySlugQuery = ( siteSlug: string ) =>
-	queryOptions( {
+export const siteBySlugQuery = ( siteSlug: string ) => {
+	// Used to find an existing Site object which is already in the `site-by-id` cache.
+	const getFromCache = () =>
+		queryClient
+			.getQueriesData< Site >( { queryKey: [ 'site-by-id' ] } )
```

Comment:
I think this must be exactly this otherwise it might get stale object whenever we update `SITE_FIELDS` or `SITE_OPTIONS` (not sure)


```suggestion
			.getQueriesData< Site >( siteByIdQuery( site.ID ) )
```